libcst/metadata/scope_provider.py

```diff
@@ -446,14 +456,6 @@ def _contains_in_self_or_parent(self, name: str) -> bool:
         """Overridden by ClassScope to hide it's assignments from child scopes."""
         return name in self
 
-    def _add_assignment(self, assignment: "BaseAssignment") -> None:
-        assignment.scope = self
```

Comment:
@zsol FYI the bug was here, we were changing assignment.scope but not assignment._index